### PR TITLE
Publish with CI

### PR DESCRIPTION
**Pull request by DifferentialOrange**
Use standard rockspec layout. Publish scm-1 rockspec with CI on push to master. Create and publish
tagged rockspecs on tag push.

Remove Travis CI build status badge, set Github Actions fast-testing and publish badge.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - 'master'
+    tags:
+      - '*'
+
+jobs:
+  publish-rockspec-scm-1:
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
```
Comment (LeonidVas) [inline .github/workflows/publish.yml:15]: `actions/checkout@v3`

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      - uses: tarantool/rocks.tarantool.org/github-action@master
+        with:
+          auth: ${{ secrets.ROCKS_AUTH }}
+          files: mysql-scm-1.rockspec
+
+  publish-rockspec-tag:
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
```
Comment (LeonidVas) [inline .github/workflows/publish.yml:25]: `actions/checkout@v3`

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      - uses: tarantool/setup-tarantool@v1
+        with:
+          tarantool-version: '2.10'
+
+      - run: echo "TAG=${GITHUB_REF##*/}" >> $GITHUB_ENV
```
Comment (LeonidVas) [inline .github/workflows/publish.yml:30]: I remember that @Totktonada  didn't like such method and advised to use https://github.com/tarantool/queue/blob/e56023dc80d77b369cdccb135146fd1f34e770de/.github/workflows/publish.yml#L25 (I don't remember the details). I would suggest comparing the results and making a decision.

**Review by LeonidVas — CHANGES_REQUESTED** [review]
Hi! Thank you for the patchset.
`ci: publish rockspecs` -> `cd: publish rockspecs`
See several comments bellow.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (DifferentialOrange) [reply .github/workflows/publish.yml]: Used v3 in existing and new pipelines

File: .github/workflows/publish.yml (hunk 2 above)
Comment (DifferentialOrange) [reply .github/workflows/publish.yml]: Used v3 in existing and new pipelines

File: .github/workflows/publish.yml (hunk 3 above)
Comment (DifferentialOrange) [reply .github/workflows/publish.yml]: Manually sed-ed version is actually the same as scm-1 with replaced `"version"` and `"tag"`, as expected.

`new_version` template is actually a bit weird: it uses three spaces for indentation. And it loses all additions like `--vim` directive comment. And it maybe won't be compatible with non-cmake rockspecs. Let's stick to the queue version.

**Review by LeonidVas — APPROVED** [review]
LGTM

**Pull request merged**